Sprint1 : combine PR for visualization from Sprint1

abasatwar opened the pull request:
### Description
Visualization implementatationfor Time series, Heatmap, Pie, Table view and UI chnages for config panel is included in this PR.

**Issues Resolved**
[Enhancement]: UI for dimension and metrics #797
[Enhancement]: Time Series - Visualizations config panel features v1 #794
[ENHANCEMENT]: Heatmap visualization config panel features v1 #791
[ENHANCEMENT]: Pie visualization config panel features v1 - #793
Table View Enhancement for Visualization #772

**Check List**
 - [x] New functionality includes testing.
- [x] Commits are signed per the DCO using --signoff
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

mengweieric commented on dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx:
```diff
@@ -131,16 +142,16 @@ export const DataTable = ({ visualizations, layout, config }: any) => {
-        setIsFullScreen(false);
-      }
+      setIsFullScreen(false);
+      // if (isFullScreen) {
```
Please remove commented code here

abasatwar replied:
This has been removed in later commit.

mengweieric commented:
Please use common PR template for this and moving forward. Add briefing for changes that have been made, and check items that apply. [Sample PR](https://github.com/opensearch-project/observability/pull/763)

abasatwar commented on dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx:
```diff
@@ -0,0 +1,86 @@
+        />
+      </>)
+  }
+  console.log("schemas:", schemas);
```
It is already been deleted in later commit and not present in PR.

abasatwar commented on dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx:
```diff
@@ -43,42 +29,77 @@ export const ConfigGraphStyle = ({
+    lineWidth: lineWidth.defaultState,
+    opacity: fillOpacity.defaultState
+  });
+  const onChange = (e) => {
```
This code is already updated in later commit.

abasatwar commented on dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx:
```diff
@@ -43,42 +29,77 @@ export const ConfigGraphStyle = ({
+    });
+  }, [schemas, vizState, handleConfigurationChange])
 
+  console.log("visualization:: ", visualizations)
```
This code is already updated in later commit.

abasatwar commented on dashboards-observability/public/components/visualizations/charts/lines/line.tsx:
```diff
@@ -25,10 +25,12 @@ export const Line = ({ visualizations, layout, config }: any) => {
+
+  const mode = dataConfig?.graphStyle?.style
+    ? dataConfig.graphStyle.style
+    : 'lines';
```
This code is already updated in later commit and strings are moved to constants file.

abasatwar commented on dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:
```diff
@@ -57,30 +57,32 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations, tabID }:
     let listItem = { ...list[name][index] };
     listItem = { ...listItem, [field]: value };
-    const newList = {
+    const x = {
```
This code is already updated in later commit.

abasatwar commented on dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:
```diff
@@ -114,97 +96,89 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations, tabID }:
+                placeholder="Custom label"
+                value={singleField.custom_label}
+                onChange={(e) => updateList(e.target.value, index, sectionName, 'custom_label')}
+                aria-label="Use aria labels when no actual label is in use" />
```
This is handled in later commit.

rinku-kumar-psl commented on dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx:
```diff
@@ -0,0 +1,88 @@
+
+  const dimensions = useMemo(() =>
+    currentSchemas && currentSchemas.map((schema: IConfigPanelOptionSection, index: string) => {
+      let params;
```
Thanks @mengweieric , Initialized params with default values.

abasatwar commented:
> Please use common PR template for this and moving forward. Add briefing for changes that have been made, and check items that apply. [Sample PR](https://github.com/opensearch-project/observability/pull/763)

Hi @mengweieric , Update PR as per suggested PR tempate. Also , we have addressed all your comments.
It would be great if you can provide your inputs.

Thanks,
Amruta

ps48 commented on dashboards-observability/common/constants/data_table.ts:
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const GRID_HEADER_COLUMN_MAX_WIDTH = '150px';
+export const GRID_PAGE_RANGE_DISPLAY = 5;
+export const COLUMN_DEFAULT_MIN_WIDTH = 100;
+export const GRID_PAGE_SIZES = [10, 50, 100];
+export const ROW_DENSITIES = [
+  { icon: 'tableDensityExpanded', height: 55, selected: false },
+  { icon: 'tableDensityNormal', height: 45, selected: false },
+  { icon: 'tableDensityCompact', height: 35, selected: true },
+];
+
+export const HEADER_HEIGHT = 35;
```
If the data table is used inside event analytics let's move this to `constants/explorer.ts`, if shared by other components let's move to `constants/shard.ts`

ps48 commented on dashboards-observability/package.json:
```diff
@@ -17,8 +17,11 @@
     "@nteract/outputs": "^3.0.11",
     "@nteract/presentational-components": "^3.4.3",
     "@reduxjs/toolkit": "^1.6.1",
+    "ag-grid-community": "^27.3.0",
+    "ag-grid-react": "^27.3.0",
```
Why do we need to add this new dependency? Is there an issue where we discussed what was not available in EUI components. Adding more packages, puts lot more maintenance issues in long term.  Thus, we should have strong reasons to support addition of this package.

abasatwar replied:
hi @ps48  ,
We have changed table view implementation as per [https://github.com/opensearch-project/observability/issues/772](url).And as per new implementation, new dependencies are added and we have confirmed it on call and tkt too.

thanks,
Amruta

pratibhapandey16 commented on dashboards-observability/.cypress/utils/event_constants.js:
```diff
@@ -110,3 +110,31 @@ export const renderPieChart = () => {
 };
+
+export const renderDataConfig = () => {
+  cy.get('.euiResizablePanel.euiResizablePanel--middle').contains('Data Cofigurations');
```
Sure. Going forward we will make sure of using `data-test-subj` wherever application in place of classname.
